### PR TITLE
[Windows] Respect `ClearButtonVisibility = ClearButtonVisibility.Never` for `<Entry>`s

### LINKED ISSUE #23112: Calling Focus() on an Entry befor it has been displayed causes ClearButton to be displayed permanently
### Description

If you add a Entry with `ClearButtonVisibility = ClearButtonVisibility.Never ` via code and call its `Focus()-`Methode befor the UI-Thread had time to update, the entry will always show its ClearButton. 

image

This will not occure if `ClearButtonVisibility = ClearButtonVisibility.WhileEditing`.

Under Android, it seems to work fine:
image

### Steps to Reproduce

Use the following content for your `MainPage.xaml`
```
  <ScrollView>
      <VerticalStackLayout x:Name="mainLayout" Padding="30,0" Spacing="25">
          <Button Text="Add 'Never' Entry"  Clicked="OnAddNeverEntry" HorizontalOptions="Fill" />
          <Button Text="Add 'While Editing' Entry"  Clicked="OnAddWhileEntry" HorizontalOptions="Fill" />
          <Button Text="Add 'Never' Entry - then change"  Clicked="OnAddNeverEntryThenChange" HorizontalOptions="Fill" />
          <Button Text="Add 'While Editing' Entry - then change"  Clicked="OnAddWhileEntryChange" HorizontalOptions="Fill" />
      </VerticalStackLayout>
  </ScrollView>

```

Use the following methods for your `MainPage.xaml.cs`
```
  private void OnAddNeverEntry(object sender, EventArgs e)
  {
      Entry entry = new Entry() { ClearButtonVisibility = ClearButtonVisibility.Never, Text = "Some Text"};
      mainLayout.Children.Add(entry);
      entry.Focus();
  }

  private void OnAddWhileEntry(object sender, EventArgs e)
  {
      Entry entry = new Entry() { ClearButtonVisibility = ClearButtonVisibility.WhileEditing, Text = "Some Text" };
      mainLayout.Children.Add(entry);
      entry.Focus();
  }

  private void OnAddNeverEntryThenChange(object sender, EventArgs e)
  {
      Entry entry = new Entry() { ClearButtonVisibility = ClearButtonVisibility.Never, Text = "Some Text" };
      mainLayout.Children.Add(entry);
      entry.ClearButtonVisibility = ClearButtonVisibility.WhileEditing;
      entry.Focus();
  }

  private void OnAddWhileEntryChange(object sender, EventArgs e)
  {
      Entry entry = new Entry() { ClearButtonVisibility = ClearButtonVisibility.WhileEditing, Text = "Some Text" };
      mainLayout.Children.Add(entry);
      entry.ClearButtonVisibility = ClearButtonVisibility.Never;
      entry.Focus();
  }
```

After launching your app, you can add entrys to the layout by pressing the buttons. 
Pressing the first or fourth button adds an entry with a permanent ClearButton will be displayed.
Pressing the second or third button adds an entry will be displayed where the ClearButton is only beeing displayed while editing

If you delay the call of `Focus()`, it works fine.
```
  private void OnAddNeverEntry(object sender, EventArgs e)
  {
      Entry entry = new Entry() { ClearButtonVisibility = ClearButtonVisibility.Never, Text = "Some Text"};
      mainLayout.Children.Add(entry);
      Dispatcher.DispatchDelayed(TimeSpan.FromSeconds(1), () =>
      {
          entry.Focus();
      });
  }
```

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.40 SR5

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

net8.0-windows10.0.17763.0 and net9.0-windows10.0.19041.0

### Did you find any workaround?

1. Call the `Focus()`-Methode after the entry has been displayed
2. Modify `App.xaml` under Windows to create a custom style for entrys, based on the default style https://github.com/microsoft/microsoft-ui-xaml/blob/winui3/release/1.5.1/controls/dev/CommonStyles/TextBox_themeresources.xaml, where u remove the content of VisualState "ButtonVisible". This will remove the ClearButton completly.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by MartyIX**
### Description of Change

The style interception code works sometimes but not always. 

Relevant XAML template in WinUI 3 is [here](https://github.com/microsoft/microsoft-ui-xaml/blob/98a60c8f30c84f297a175dd2884d54ecd1c8a4a9/controls/dev/CommonStyles/TextBox_themeresources.xaml#L311-L341).

### Issues Fixed

Fixes #23112
Related to #13714

PR that introduced the modified code: #3444

### CODE REVIEW

File: src/Core/src/Platform/Windows/MauiTextBox.cs [hunk 1]
```diff
@@ -60,56 +61,24 @@ static void OnVerticalTextAlignmentPropertyChanged(DependencyObject d, Dependenc
...
 			{
-				if (state.Name == ButtonVisibleStateName)
-					return (buttonStates, state);
+				if (GetIsDeleteButtonEnabled(element))
+				{
+					deleteButton.RenderTransform = null;
+				}
+				else
+				{
+					deleteButton.RenderTransform = new UI.Xaml.Media.TranslateTransform() { Y = -1000 };
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/Core/src/Platform/Windows/MauiTextBox.cs:79]: What happens if you have entries in a scroll view... Will it be a floating button somewhere?

**Comment by mattleibow** [comment]
Thanks for the test, now to bend CI to our will and ask it to do the basic job of running the tests properly.

**Comment by mattleibow** [comment]
You can download the screenshots from the devops test attachments. Just go to the failed test and the attachments tab. Should be there. It is the one without the nav bars and status bars.

**Comment by mattleibow** [comment]
I think the layout will not really work on mobile:

| Android | iOS |
| :-: | :-: |
| image | image |

https://dev.azure.com/xamarin/public/_build/results?buildId=118577&view=ms.vss-test-web.build-test-results-tab&runId=2906687&resultId=100119&paneView=attachments

But it seems to be right for Windows:

image

I wonder if it will work if you have jus a flay vertical stack layout?

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23158.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		App.Click("AddEntry");
+
+		await Task.Delay(500);
+
+		// Click the new entry to see if there is the clear button or not. No such button should be present.
+		App.Tap("Entry3");
+
+		await Task.Delay(500);
+
+		VerifyScreenshot();
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23158.cs:31]: macOS won't generate screen shots so I would just ignore this test on macOS

Do we need the 500 ms delays?

VerifyScreenshot already does a delay and retry

**Review by Foda — APPROVED** [review]

**Comment by mattleibow** [comment]
Trying the tests on Windows Core unpackaged again... Probably not related since the packaged version ran. But just to be safe. Might just merge if it fails again and see if main fixes it.

**Comment by mattleibow** [comment]
Well, now it is green so whatever CI.

**Review by mattleibow — APPROVED** [review]

**Pull request merged**